validator/sawtooth_validator/gossip/gossip_handlers.py

```diff
@@ -156,30 +156,72 @@ def handle(self, connection_id, message_content):
 
 
 class GossipBlockResponseHandler(Handler):
+    def __init__(self, completer, responder, chain_controller_has_block):
+        self._completer = completer
+        self._responder = responder
+        self._chain_controller_has_block = chain_controller_has_block
+
     def handle(self, connection_id, message_content):
-        ack = NetworkAcknowledgement()
-        ack.status = ack.OK
         block_response_message = GossipBlockResponse()
         block_response_message.ParseFromString(message_content)
+        block = Block()
+        block.ParseFromString(block_response_message.content)
+
+        block_id = block.header_signature
+
+        if not self._has_open_requests(block_id) and self._has_block(block_id):
+            LOGGER.debug('Drop duplicate block: %s', block_id)
+            return HandlerResult(HandlerStatus.DROP)
```

Comment:
This should return a Network Ack to close the future from the sender. HandlerStatus.RETURN will return the Ack but stop processing of the message. Same for below.